### PR TITLE
PKCS12 interface

### PR DESCRIPTION
**Pull request by reaperhulk**
refs #4553

### CODE REVIEW

**Comment by reaperhulk** [comment]
I suppose this fails coverage because nothing imports it. Well, we can still discuss.

File: src/cryptography/hazmat/primitives/serialization/pkcs12.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+from __future__ import absolute_import, division, print_function
+
+import abc
+
+import six
+
+
+@six.add_metaclass(abc.ABCMeta)
+class PKCS12(object):
+    @abc.abstractmethod
```
Comment (alex) [inline src/cryptography/hazmat/primitives/serialization/pkcs12.py]: I don't think this should be on the interface

**Comment by reaperhulk** [comment]
I haven't been able to come up with a great solution. The PKCS12 spec is just a pile of bags. These getters are a subset (e.g. `private_key` is really Union(KeyBag, PKCS8ShroudedKeyBag), `certificate` is a CertBag that matches the private key, and the additional certificates are any other CertBag), so it's possible we could just evolve the interface in the future we choose to support more.

Do we know of anything that actually implements a significantly larger chunk of this spec? Go supports just cert + key for reference.

**Comment by reaperhulk** [comment]
(We're going to go with a function returning a tuple to avoid creating an interface we don't need yet)

**Pull request closed without merging**